airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py

```diff
@@ -350,6 +351,7 @@ def read_records(
 class IncrementalFileStream(FileStream, ABC):
     # TODO: ideally want to checkpoint after every file or stream slice rather than N records
     state_checkpoint_interval = None
+    days = 3  # keeping track of all files synced in the last N days
```

Comment:
@sherifnada There might be a lot of files synced, is there any limit to the size that `state` can be?

Reply:
Should this be an optional configuration for how long to store?

Reply:
there's no advertised limit ATM. Asked the platform team. Some alternatives I can think of to make this smaller: 
1. hash each filename to a fixed size as a compression mechanism
2. compress the list of filenames using some compression algo that that can go in a JSON object (i.e; represnted using ascii)
3. we can start to do "interesting" things like factor out common file prefixes etc.. to save space too

Reply:
the limit is currently 1GB. I feel like if this object ever gets close to that, we should just set a flag like `sync_all_files_always` to true and just always sync all files from the last `buffer` days

Reply:
when history size is more than 1000000000 bytes (0.93 GB) we delete history, `sync_all_files_always` flag switched to true and we always sync all files from the last `buffer days`.